tests/test_parser_grimoirelab.py

```diff
@@ -0,0 +1,342 @@
+        self.assertEqual(dom.domain, 'gsyc.es')
+        self.assertEqual(dom.is_top_domain, False)
+
+    def test_enrollments_parser(self):
+        """Check whether enrollments are correctly parsed"""
+
+        stream_ids = self.read_file('data/grimoirelab_valid.yml')
+
+        parser = GrimoireLabParser(stream_ids)
+
+        # Parsed unique identities
+        uids = parser.identities
+
+        # J. Manrique Lopez de la Fuente
+        uid = uids[0]
+        rol = uid.enrollments[0]
+        self.assertIsInstance(rol, Enrollment)
+        self.assertEqual(rol.organization.name, 'Bitergia')
+        self.assertEqual(rol.start, datetime.datetime(2013, 1, 1, 0, 0))
+        self.assertEqual(rol.end, datetime.datetime(2100, 1, 1, 0, 0))
+
+        # Luis Cañas-Díaz
+        uid = uids[1]
+        rol = uid.enrollments[0]
+        self.assertIsInstance(rol, Enrollment)
+        self.assertEqual(rol.organization.name, 'GSyC/LibreSoft')
+        self.assertEqual(rol.start, datetime.datetime(2003, 1, 1, 0, 0))
+        self.assertEqual(rol.end, datetime.datetime(2011, 12, 31, 0, 0))
+
+        rol = uid.enrollments[1]
+        self.assertIsInstance(rol, Enrollment)
+        self.assertEqual(rol.organization.name, 'Bitergia')
+        self.assertEqual(rol.start, datetime.datetime(2012, 1, 1, 0, 0))
+        self.assertEqual(rol.end, datetime.datetime(2100, 1, 1, 0, 0))
+
+    def test_not_valid_organizations_stream(self):
+        """Check whether it parses invalid organizations files"""
+
+        # empty domains
+        stream_orgs = self.read_file('data/grimoirelab_orgs_invalid_empty_domains.yml')
+        with self.assertRaises(InvalidFormatError):
+            GrimoireLabParser(domain_employer=stream_orgs)
+
+        # one of the domains is empty
+        stream_orgs = self.read_file('data/grimoirelab_orgs_invalid_domains_list_with_empty_value.yml')
+        with self.assertRaises(InvalidFormatError):
+            GrimoireLabParser(domain_employer=stream_orgs)
+
+        # domains got a string instead of a list
+        stream_orgs = self.read_file('data/grimoirelab_orgs_invalid_wrong_domains_type.yml')
+        with self.assertRaises(InvalidFormatError):
+            GrimoireLabParser(domain_employer=stream_orgs)
+
+        # organization key missing
+        stream_orgs = self.read_file('data/grimoirelab_orgs_invalid_missing_key.yml')
+        with self.assertRaises(InvalidFormatError):
+            GrimoireLabParser(domain_employer=stream_orgs)
+
+        # organization key with empty value
+        stream_orgs = self.read_file('data/grimoirelab_orgs_invalid_key_with_no_value.yml')
+        with self.assertRaises(InvalidFormatError):
+            GrimoireLabParser(domain_employer=stream_orgs)
+
+    def test_not_valid_identities_stream(self):
+        """Check whether it parses invalid identities files"""
+
+        stream_ids = self.read_file('data/grimoirelab_invalid_email.yml')
+        with self.assertRaises(InvalidFormatError):
```

Comment:
Check these exceptions with `assertRaisesRegExp` so we can know for sure whether is raising the right exception.